tests/acceptance/services/lazy-loader-test.js

```diff
@@ -151,3 +151,39 @@ test('dependsOn doesnt break for circular dependencies', function(assert) {
   assert.notEqual(assets2.indexOf('bundleA'), -1);
   assert.notEqual(assets2.indexOf('bundleB'), -1);
 });
+
+test('dependsOn respects the order', function(assert) {
+  const service = getSubject();
+  service.setBundles([{
+    name: '1',
+    dependsOn: ['2'],
+  }, {
+    name: '2',
+    dependsOn: ['4', '3'],
+  }, {
+    name: '3',
+    dependsOn: ['4'],
+  }, {
+    name: '4'
+  }]);
+
+  var assets = namesForBundles(service.getDependentBundlesForBundle(service.getBundleByName('1')));
```

Comment:
seems like this should be two separate tests. also "dependsOn respects the order" as the name is confusing when you swap the order of the dependsOn but the results don't change